Document backwards incompatible `kube_job` tag in KSM core (#13808)
DataDog/documentation: content/en/integrations/kubernetes_state_core.md

`kubernetes_state.container.waiting` and `kubernetes_state.container.status_report.count.waiting`
: These metrics no longer emit a 0 value if no pods are waiting. They only report non-zero values.

`kube_job`

> : The `kube_job` tag in `kubernetes_state` contained the `CronJob` name if the `Job` had that as an owner, and otherwise it contained the `Job` name itself. In `kubernetes_state_core`, `kube_job` always contains the `Job` name itself, and a new `kube_cronjob` tag contains the `CronJob` name. When migrating to `kubernetes_state_core`, it's recommended to use the new tag, or by using `kube_job:foo*` (where `foo` is the `CronJob` name) in queries for filtering.

Reviewer: ```suggestion
: The `kube_job` tag value in `kubernetes_state` is the `CronJob` name if the `Job` had that as an owner, otherwise it is the `Job` name. In `kubernetes_state_core`, the `kube_job` tag value is always the `Job` name, and a new `kube_cronjob` tag key is added with the `CronJob` name as the tag value. When migrating to `kubernetes_state_core`, it's recommended to use the new tag or `kube_job:foo*`, where `foo` is the `CronJob` name, for query filters.
```

Reviewer: What does "if the `Job` had that as an owner" mean? If the job had what as the owner?

Author: If the `Job` had the `CronJob` as an owner.
